RxSwift/Traits/Single.swift

```diff
@@ -99,6 +101,15 @@ extension PrimitiveSequenceType where TraitType == SingleTrait {
                 }
             }
         }
+
+        if let disposed = onDisposed {
```

Comment:
I don't think this is gong to do what you want. As just one case, consider:
```
Just(1).asSingle().subscribe(onSuccess: { _ in }).disposed(by: bag)
```
The call to self.primitiveSequence.subscribe is going to fire the subscription to Just, return the value, complete and dispose all before this line is even hit. As a result, the subscription is already disposed, but your onDispose handler isn't going to be fired until the disposeBag is deallocated and emptied.

If you're going to do this, you're going to need to hook into `self.primitiveSequence.subscribe` much in the same way as it's done in the subscribe method in `ObservableType+Extensions.swift`.

That would probably also require that you add onDisposed to the other traits as well.

Reply:
Oh, good point. I'll wait for some feedback on whether I should continue with the PR or scrap it altogether.